index.js

```diff
@@ -68,6 +68,15 @@ module.exports = {
   included: function() {
     this._super.included.apply(this, arguments);
 
+    let isApp = !this.project.isEmberCLIAddon();
+
+    let hasEmberFetch = this.project.addons.map(addon => addon.name).includes('ember-fetch');
```

Comment:
This looks good, but there is a more concise API that could be used:

`project.findAddonByName('ember-fetch')` https://github.com/ember-cli/ember-cli/blob/807f56e2e434f716757de02fd1f2c3251dadaf6f/lib/models/project.js#L634

I noticed this method was listed as private, but it's intent was to be public so I've fixed that:https://github.com/ember-cli/ember-cli/pull/8621

Reply:
Another approach, would be to do this work in the `init()` would be via `this.project.dependencies['ember-fetch']`

This approach would allow us to disable nested `ember-fetch` instances via `isEnabled() { }` hook, whereas if `isEnabled` returns false, we won't hit the `included` hook.

Reply:
I made the change to use `findAddonByName` API for checking if addon exists. Also, updated the condition to check if the app has ember-cli-fastboot. If that addon exists and ember-fetch is not included throw the error.

Reply:
> Another approach, would be to do this work in the `init()` would be via `this.project.dependencies['ember-fetch']`
> 
> This approach would allow us to disable nested `ember-fetch` instances via `isEnabled() { }` hook, whereas if `isEnabled` returns false, we won't hit the `included` hook.

I am just wondering, using above and disabling nested fetch can have any side-effects.